internal/os/smb/api.go

```diff
@@ -14,8 +15,40 @@ func New() APIImplementor {
 	return APIImplementor{}
 }
 
-func (APIImplementor) NewSmbGlobalMapping(remotePath, username, password string) error {
-	klog.V(4).Infof("NewSmbGlobalMapping: remotePath:%q", remotePath)
+func (APIImplementor) IsSmbMapped(remotePath string) (bool, error) {
+	cmdLine := fmt.Sprintf(`$(Get-SmbGlobalMapping -RemotePath $Env:smbremotepath -ErrorAction Stop).Status `)
+	cmd := exec.Command("powershell", "/c", cmdLine)
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("smbremotepath=%s", remotePath))
+
+	out, err := cmd.CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("error checking smb mapping %s, err: %v", remotePath, err)
+	}
+
+	if len(out) == 0 || !strings.EqualFold(strings.TrimSpace(string(out)), "OK") {
+		return false, nil
+	}
+	return true, nil
+}
+
+func (APIImplementor) SMBLink(remotePath, localPath string) error {
```

Comment:
this is a generic way to create symlink, do we want to only put it in smb api?

Reply:
Please put a comment here explaining some of the subtle aspects around links that you found out when looking into this and why we need a special mechanism for SMB.